src/Microsoft.OData.Client/Util.cs

```diff
@@ -597,6 +597,18 @@ internal static Version ToVersion(this ODataProtocolVersion protocolVersion)
             return parameters.ToDictionary(parameter => parameter.Name, parameter => Serializer.GetParameterValue(context, parameter));
         }
 
+        /// <summary>
+        /// Quick check if the <paramref name="input"> is Json.
+        /// Note that this does not check if the input is a well formed Json.
+        /// This can be used to check verify that the string is not xml
+        /// </summary>
+        /// <param name="input">the input string.</param>
+        /// <returns>Return true if the input string can be Json.</returns>
+        public static bool CanBeJson(string input)

```

Comment:
No need to parse Json at this point just to check if the string is not XML. Remember OData response messages can be JSON or XML. This is a simple/Lazy check to eliminate possibility of the string being XML.

Reply:
Since this check does not guarantee that the payload is indeed json, or that parsing will succeed, why not just let it be parsed as json, and if it's not, an exception will be thrown? Because in this case an exception will still be thrown if the json is not valid.

Reply:
At this point in time we do not need to parse the whole string. We only need to eliminate the fact that the response is from the service is not in `OData Atom` format. OData service response can either be `JSON `or `OData Atom` format.